LabelContainer.tsx

```diff
@@ -1,25 +1,17 @@
-import React, { PureComponent } from 'react';
-import { View } from 'react-native';
+import React, { useState, ReactNode } from 'react';
+import { StyleProp, View, ViewProps } from 'react-native';
 
-class LabelContainer extends PureComponent {
+export interface Props<T> {
+  renderContent: (value: T) => ReactNode;
+}
 
-  state = {
-    value: Number.NaN,
-  };
-  
-  setValue = value => {
-    this.setState({ value });
-  }
+const LabelContainer = <T extends unknown>({
+  renderContent,
+  ...restProps
+}: Props<T> & StyleProp<ViewProps>) => {
+  const [value, setValue] = useState<T | typeof Number.NaN>(Number.NaN);
```

Comment:
`setValue` is a method which is called on this component's ref [here](https://github.com/githuboftigran/rn-range-slider/blob/5b0c535f1ac98b8d8cef07ee9ca17a7cc55a29f8/hooks.tsx#L118).
It can't be a state variable.
The reason is described [here](https://github.com/githuboftigran/rn-range-slider/blob/5b0c535f1ac98b8d8cef07ee9ca17a7cc55a29f8/hooks.tsx#L89).